### PR TITLE
Issue #11985: sarif formatter option added to ant task

### PR DESCRIPTION
**Pull request by jknair0**
#11985

1. created an uber jar using `mvn -P assembly package`
2. used the resultant jar along with the code in ant project - https://github.com/jknair0/AntCheckStyleTest/tree/pr/checkstyle-sarif-test
3. ran `ant checkstyle`
4. It generates a valid 'sarif' report.

### CODE REVIEW

**Comment by rnveach** [comment]
Please look over CI failures.

````
[WARNING] Rule violated for class com.puppycrawl.tools.checkstyle.ant.CheckstyleAntTask.Formatter: lines covered ratio is 0.96, but expected minimum is 1.00
[WARNING] Rule violated for class com.puppycrawl.tools.checkstyle.ant.CheckstyleAntTask.Formatter: branches covered ratio is 0.95, but expected minimum is 1.00
````

**Comment by romani** [comment]
Please rebase on our latest master code

**Comment by jknair0** [comment]
Looks like all checks passed now. Can you please point out if I am wrong somewhere. I am not sure if I am missing something to check in Jobs.

**Review by romani — CHANGES_REQUESTED** [review]
Please add this format at https://github.com/checkstyle/checkstyle/blob/a368230b7df8ca70839f97d30a8d8adebd8b9ff3/config/ant-phase-verify.xml#L39 we need to eat food we produce

**Comment by romani** [comment]
Please update our documentation
https://github.com/checkstyle/checkstyle/blob/master/src/xdocs/anttask.xml.vm

File: src/xdocs/anttask.xml.vm [hunk 1]
```diff
@@ -402,14 +406,15 @@
       <p>
         <b>
           Run checkstyle on a set of files and output messages to standard
-          output in plain format, and a file in XML format
+          output in plain format, and files in XML &amp; Sarif format
```
Comment (romani) [inline src/xdocs/anttask.xml.vm:409]: Uppercase please - SARIF

>SARIF, the Static Analysis Results Interchange Format

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/test/java/com/puppycrawl/tools/checkstyle/ant/CheckstyleAntTaskTest.java [hunk 2]
```diff
@@ -527,6 +528,34 @@ public final void testXmlOutput() throws IOException {
...
+        antTask.addFormatter(formatter);
+        antTask.execute();
+
+        final List<String> expected = readWholeFile(
+                new File(getPath("ExpectedCheckstyleAntTaskSarifOutput.sarif")));
+        final List<String> actual = readWholeFile(outputFile);
+        for (int lineNumber = 0; lineNumber < expected.size(); lineNumber++) {
+            final String line = expected.get(lineNumber);
+            if (line.trim().startsWith("\"uri\"")) {
+                continue;
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/ant/CheckstyleAntTaskTest.java:551]: Skipping is not good approach.
Can create sub string after `**` of expected line and do contains in actual

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/test/java/com/puppycrawl/tools/checkstyle/ant/CheckstyleAntTaskTest.java (hunk 2 above)
Comment (jknair0) [reply src/test/java/com/puppycrawl/tools/checkstyle/ant/CheckstyleAntTaskTest.java]: Not sure how the behaviour would be in win. Do we have a generic path comparing method in this project or Do you have a suggestion on how this has to be handled ?

Comment (romani) [reply src/test/java/com/puppycrawl/tools/checkstyle/ant/CheckstyleAntTaskTest.java]: I am open for suggestions. I I just pointing that just ignoring is not very good idea.
Some  partial validation would be good.

Comment (jknair0) [reply src/test/java/com/puppycrawl/tools/checkstyle/ant/CheckstyleAntTaskTest.java]: Normalized the windows path before comparing.

**Review by romani — APPROVED** [review]
Ok to merge

**Comment by rnveach** [comment]
GitHub, rebase

**Comment by rnveach** [comment]
I have no issues to merge, I wanted to make sure CI will pass on the latest commit since the current one seemed a bit old.

**Comment by romani** [comment]
@jknair0 , please rebase on latest

**Comment by rnveach** [comment]
I had to approve a full CI run. I guess it was never approved before.

**Review by rnveach — APPROVED** [review]

**Pull request merged**